pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1262,6 +1262,20 @@ func (c *Cluster) startMon(m *monConfig, schedule *MonScheduleInfo) error {
 		logger.Debugf("adding host path volume source to mon deployment %s", d.Name)
 	}
 
+	// Expand mon PVC if storage request for mon has increased in cephcluster crd
+	if pvcExists && c.monVolumeClaimTemplate(m) != nil {
```

Comment:
How about moving the PVC expansion inside the `pdateMon()` method, immediately before the mon deployment is updated? That will help keep this long method from growing more.

Reply:
IMO updateMon is totally different than pvcExpand. If we do want to keep the startMon small i think we should create some more helper funcs

Reply:
moreover, PVC Expand should be done before monUpdate. So that PVC expansion is done before mon update and mon can utilize the new size.

Reply:
That's the point, the PVC will only ever need to be updated if the mon deployment already exists. So if we update the mon PVC and then update the mon deployment, those actions naturally go together, right?

Reply:
Done